src/math/Vector4.js

```diff
@@ -247,6 +247,17 @@ Object.assign( Vector4.prototype, {
 
 	},
 
+	multiply: function ( v ) {
+
+		this.x *= v.x;
+		this.y *= v.y;
+		this.z *= v.z;
+		this.w *= v.w || 1;
```

Comment:
Why would v.w be undefined?

What happens when v.w is 0?

Can you please explain your use case for this method?

Reply:
Currently I am creating 4D model. Every vertice of my model have 4 coordinate. I have modified the THREE.BufferGeometry for my 4D model:

```
Object.assign( THREE.BufferGeometry.prototype, {

	setFromPoints: function ( points, itemSize ) {

		itemSize = itemSize || 3;
		var position = [];

		for ( var i = 0, l = points.length; i < l; i++ ) {

			var point = points[i];
			position.push( point.x, point.y, point.z || 0 );
			if ( itemSize >= 4 )
				position.push( point.w || 0 );

		}

		this.addAttribute( 'position', new THREE.Float32BufferAttribute( position, itemSize ) );

		return this;

	},

} );
```

Now geometry.attributes.position.itemSize = 4 and every vertice is THREE.Vector4.

Sometimes I want to scale my 4D model in 3D space. I have multiple every 4D point of my 4D model by 3D scale:
`new THREE.Vector4().fromArray( mesh.geometry.attributes.position.array, i * mesh.geometry.attributes.position.itemSize ).multiply(mesh.scale)`
i is point index
mesh is my 4D model.
mesh.scale.w is undefined, then v.w = undefined and `this.w *= 1`.
if v.w is 0 then `this.w = 0`.

Reply:
>if v.w is 0 then this.w = 0.

As you have written the method, if `v.w` is 0, then `this.w` is unchanged.

I think your use case is application-specific.

Reply:
Sorry. See [Vector4.multiply bug](https://github.com/mrdoob/three.js/pull/19495/commits/5983029600edc2ef46ecfaa104bf4b89fc06b53b)

Reply:
I'll find this implementation confusing. It would implement this method similar to `add()` or `sub()` so just:
```js
multiply: function ( v ) {

    this.x *= v.x;
    this.y *= v.y;
    this.z *= v.z;
    this.w *= v.w;

    return this;

},
```

Reply:
@Mugen87 @mrdoob Your suggested change is appropriate, but it will not be acceptable to the OP because he is passing in `v.w` as `undefined` in his app.

I cannot think of a use case for this method. Maybe that is why there has been no demand for it in 10 years.

Reply:
>  but it will not be acceptable to the OP because he is passing in v.w as undefined in his app.

Yeah, but unfortunately the suggested version of the OP is too application specific. 

@anhr It seems @mrdoob supports the addition of `Vector4.multiply()` in the way I have outlined here: https://github.com/mrdoob/three.js/pull/19495#discussion_r432559787. This is the only way we can support `Vector4.multiply()`. If the implementation adds no benefit for your use case, it's probably better to close the PR until more users request the method.

Reply:
>I cannot think of a use case for this method. Maybe that is why there has been no demand for it in 10 years.
> If the implementation adds no benefit for your use case, it's probably better to close the PR until more users request the method.

That is what I would recommend, too, but I also know it is a judgment call. :-)